src/clients/java/src/jni.zig

```diff
@@ -0,0 +1,3230 @@
+///! Based on Java Native Interface Specification and Java Invocation API.
+///! https://docs.oracle.com/en/java/javase/17/docs/specs/jni.
+const std = @import("std");
```

Comment:
Could you add a motivation here? Basically, that because interface is stable and pinned down by java langauge specification, that jni.h has an unfortunate license, and so we just "manually" code this and rely on exhaustive tests to catch out any bugs?

Reply:
Those are valid points, but I think it's not only about that.

If you inspect the very own `jni.h` [C++ declarations](https://github.com/openjdk/jdk/blob/ec7da91bd83803b7d91a4de3a01caf0ba256c037/src/java.base/share/native/include/jni.h#L780-L811), you'll see it's equally error-prone.

I mean, nothing prevents a human error from calling the wrong vtable function or passing the wrong argument.
In that respect, even if we distribute the .h file or a translated version of it, we're still in the same situation when our API needs to call the vtable pointer (like the C++ API does).

That's why I implemented the vtable using only indexes to minimize the amount of code we need to maintain.

```C++
/*
 * We use inlined functions for C++ so that programmers can write:
 *
 *    env->FindClass("java/lang/String")
 *
 * in C++ rather than:
 *
 *    (*env)->FindClass(env, "java/lang/String")
 *
 * in C.
 */

struct JNIEnv_ {
    const struct JNINativeInterface_ *functions;
#ifdef __cplusplus

    jint GetVersion() {
        return functions->GetVersion(this);
    }
    jclass DefineClass(const char *name, jobject loader, const jbyte *buf,
                       jsize len) {
        return functions->DefineClass(this, name, loader, buf, len);
    }
    jclass FindClass(const char *name) {
        return functions->FindClass(this, name);
    }
    jmethodID FromReflectedMethod(jobject method) {
        return functions->FromReflectedMethod(this,method);
    }
    jfieldID FromReflectedField(jobject field) {
        return functions->FromReflectedField(this,field);
    }

    // ...
}
```

Reply:
I mean that we need to include our motivation as a source code comment `//!`, so that the next person doesn’t need to wonder why we do it this way. I agree that what we do is the right approach.